BedrockServer.cpp

```diff
@@ -567,11 +613,13 @@ void BedrockServer::sync(const SData& args,
                     }
 
                     // If we just started a new HTTPS request, save it for later.
+                    // WTF can this ever be reached? maybe peekCommand creates an httpsRequest?
                     if (command->httpsRequests.size()) {
                         server.waitForHTTPS(move(command));
 
                         // Move on to the next command until this one finishes.
                         core.rollback();
+                        SDEBUG("Unlocking the _syncThreadCommitMutex after new https request");
```

Comment:
Can you explain why we should take them out? They were useful in pinpointing the problem with the mutex, which I'm not confident is permanently resolved.

Reply:
If we left in every debug logline that was used to debug a weird race condition, there's literally be 5,000 debug lines in the form of `TYLER this thing happened` littering the entire code base.

Reply:
So, how do you decide whether a debug flag is useful to keep? The crashes that led to this investigation were easy enough to debug, but finding where the mutex had previously been locked was not. On the other hand, I think we're likely to have fixed these lock issues by moving the code block that handles these failures, so it might make sense to wait. I'm ambivalent, but lean towards verbose logging. Does it matter how many debug log lines there are?

Reply:
Currently there are only 39 of them and only a handful touched in 2020, and I think most of those are changes to existing lines, not new ones added. Which is to say we barely use them currently.

Loglines that are only recorded in dev are only useful in dev, in which case they're quite easy to add when debugging a specific problem.

Like I said, we'd have thousands of these if I kept every logline I added while debugging every issue. At what point does the signal/noise ratio get so low that we rip out all of the debug log lines for the last 30 bugs that we investigated? How would you analyze the log lines you added here if every mutex lock/unlock logged a debug line?

```
65309862 (Tyler Karaszewski 2017-09-08 08:09:00 -0700 246)         SDEBUG("Received command #" << (int)packet.sequenceID << ": '" << SToHex(packet.serialize()) << "'");
9ce19c33 (Tyler Karaszewski 2020-02-26 14:39:30 -0800 1675)             SDEBUG("Plugin " << pair.first << " handling command " << command->request.methodLine);
551cbbea (Tyler Karaszewski 2016-09-29 13:53:22 -0700  96)                 SDEBUG("Connect to '" << socket->addr << "' failed with SO_ERROR #" << result << ", closing.");
551cbbea (Tyler Karaszewski 2016-09-29 13:53:22 -0700 103)             SDEBUG("Connect to '" << socket->addr << "' succeeded.");
551cbbea (Tyler Karaszewski 2016-09-29 13:53:22 -0700 119)                 SDEBUG("sslState=" << SSSLGetState(socket->ssl) << ", canrecv=" << SFDAnySet(fdm, socket->s, SREADEVTS)
64c3c924 (Tyler Karaszewski 2017-03-16 12:11:37 -0700 141)                 SDEBUG("Connection to '" << socket->addr << "' died (recv=" << aliveAfterRecv << ", send="
551cbbea (Tyler Karaszewski 2016-09-29 13:53:22 -0700 160)                         SDEBUG("Graceful shutdown of SSL socket '" << socket->addr << "'");
551cbbea (Tyler Karaszewski 2016-09-29 13:53:22 -0700 186)                         SDEBUG("Graceful shutdown of socket '" << socket->addr << "'");
64c3c924 (Tyler Karaszewski 2017-03-16 12:11:37 -0700 205)     SDEBUG("Shutting down socket '" << socket->addr << "' (" << how << ")");
0214b0b2 (Tyler Karaszewski 2019-01-18 09:51:40 -0400 213)     SDEBUG("Closing socket '" << socket->addr << "'");
ecda922d (Tyler Karaszewski 2016-10-14 11:17:40 -0700 62)             SDEBUG("Accepting socket from '" << addr << "' on port '" << port.host << "'");
c2e92007 (Tyler Karaszewski 2017-09-07 18:16:16 -0700 170)                 SDEBUG("Incoming connection failed from '" << socket->addr << "' (" << e.what() << "), empty buffers");
c2e92007 (Tyler Karaszewski 2017-09-07 18:16:16 -0700 72)         SDEBUG("Failed to deserialize JSON-encoded SQResult (" << e.what() << "): " << json);
55608659 (Ionatan Wiznia    2018-03-14 14:08:33 +0100 278) #define SDEBUG(_MSG_) SSYSLOG(LOG_DEBUG, "[dbug] " << SLOGPREFIX << _MSG_)
b0b7cce8 (Tyler Karaszewski 2020-03-31 10:30:05 -0700   4) #define PDEBUG(_MSG_) SDEBUG("->{" << peer->name << "} " << _MSG_)
551cbbea (Tyler Karaszewski   2016-09-29 13:53:22 -0700 1253)                 SDEBUG("Expecting: '" << _CH_ << "', found 'eol'");                                                    \
551cbbea (Tyler Karaszewski   2016-09-29 13:53:22 -0700 1255)                 SDEBUG("Expecting: '" << _CH_ << "', found '" << ptr << "'");                                          \
551cbbea (Tyler Karaszewski   2016-09-29 13:53:22 -0700 1352)             SDEBUG("Parsed: '" << name << "':'" << value << "'");
b4989cf7 (Tyler Karaszewski   2017-02-16 11:59:21 -0800 2449)         SDEBUG(sql);
551cbbea (Tyler Karaszewski  2016-09-29 13:53:22 -0700  581)         SDEBUG("Connected to " << numLoggedInFullPeers << " of " << numFullPeers << " full peers (" << peerList.size()
1aa76dd7 (Tyler Karaszewski  2017-05-08 16:34:46 -0700  715)                             SDEBUG("Peer '" << peer->name << "' has approved transaction.");
3c76b6aa (Tyler Karaszewski  2017-03-20 12:19:04 -0700  751)             SDEBUG(     "numFullPeers="           << numFullPeers
c918f743 (Tyler Karaszewski  2018-01-16 16:14:09 -0800  782)                 SDEBUG("Committing current transaction because consistentEnough: " << _db.getUncommittedQuery());
a29fc8c2 (Mike Brannigan     2018-04-27 11:35:21 -0700 1795)         SDEBUG("Setting state timeout of " << timeout / 1000 << "ms");
c918f743 (Tyler Karaszewski  2018-01-16 16:14:09 -0800 1989)         SDEBUG("Committing current transaction because _recvSynchronize: " << _db.getUncommittedQuery());
5646707b (Tyler Karaszewski  2020-03-11 14:29:12 -0700 2277)     SDEBUG("Committing current transaction because COMMIT_TRANSACTION: " << _db.getUncommittedQuery());
8f568f10 (Tyler Karaszewski 2020-03-25 12:56:10 -0700  276)         SDEBUG("[checkpoint] starting thread with count: " << object->_sharedData->_currentPageCount.load());
006d1244 (Tyler Karaszewski 2017-01-24 18:24:42 -0300  416)     SDEBUG("Beginning transaction");
f901a3ec (Tyler Karaszewski 2017-01-25 13:04:18 -0300  447)     SDEBUG("[concurrent] Beginning transaction");
551cbbea (Tyler Karaszewski 2016-09-29 13:53:22 -0700  711)     SDEBUG("Prepared transaction");
551cbbea (Tyler Karaszewski 2016-09-29 13:53:22 -0700  745)     SDEBUG("Committing transaction");
4e3415f8 (Tyler Karaszewski 2018-01-31 12:40:06 -0800  777)         SDEBUG("Commit successful (" << _sharedData->_commitCount.load() << "), releasing commitLock.");
551cbbea (Tyler Karaszewski 2016-09-29 13:53:22 -0700  919)     SDEBUG("Getting commits #" << fromIndex << "-" << toIndex);
56d3b882 BedrockCore.cpp (Tyler Karaszewski 2020-01-30 15:32:14 -0800  78)         SDEBUG("Peeking at '" << request.methodLine << "' with priority: " << command->priority);
49fd663a BedrockCore.cpp (Tyler Karaszewski 2020-02-26 14:42:31 -0800  95)             SDEBUG("Plugin '" << command->getName() << "' peeked command '" << request.methodLine << "'");
73999b2b BedrockCore.cpp (Cole              2018-10-18 06:49:36 -0700 181)         SDEBUG("Processing '" << request.methodLine << "'");
49fd663a BedrockCore.cpp (Tyler Karaszewski 2020-02-26 14:42:31 -0800 205)                 SDEBUG("Plugin '" << command->getName() << "' processed command '" << request.methodLine << "'");
551cbbea (Tyler Karaszewski           2016-09-29 13:53:22 -0700 285)         SDEBUG("Resetting database");
551cbbea (Tyler Karaszewski           2016-09-29 13:53:22 -0700 293)         SDEBUG("Verifying database exists");
```